com.unity.render-pipelines.core/Editor/Volume/VolumeComponentProvider.cs

```diff
@@ -47,35 +48,35 @@ public VolumeComponentProvider(VolumeProfile target, VolumeComponentListEditor t
             m_TargetEditor = targetEditor;
         }
 
-        public void CreateComponentTree(List<Element> tree)
+        private static Dictionary<string, Type> FilterVolumeComponentTypes(Type[] types, Type currentPipelineType)
         {
-            tree.Add(new GroupElement(0, "Volume Overrides"));
-
-            var types = VolumeManager.instance.baseComponentTypeArray;
-            var rootNode = new PathNode();
-
+            var volumes = new Dictionary<string, Type>();
             foreach (var t in types)
             {
-                // Skip components that have already been added to the volume
-                if (m_Target.Has(t))
-                    continue;
-
                 string path = string.Empty;
 
-                // Look for a VolumeComponentMenu attribute
                 var attrs = t.GetCustomAttributes(false);
 
                 bool skipComponent = false;
+
+                // Look for the attributes of this volume component and decide how is added and if it needs to be skipped
                 foreach (var attr in attrs)
                 {
-                    if (attr is VolumeComponentMenu attrMenu)
-                        path = attrMenu.menu;
-
-                    if (attr is HideInInspector attrHide)
-                        skipComponent = true;
-
-                    if (attr is ObsoleteAttribute attrDeprecated)
-                        skipComponent = true;
+                    switch (attr)
+                    {
+                        case VolumeComponentMenu attrMenu:
+                        {
+                            path = attrMenu.menu;
+                            break;
+                        }
+                        case HideInInspector attrHide:
+                        case ObsoleteAttribute attrDeprecated:
+                            skipComponent = true;
+                            break;
+                        case SupportedOnAttribute supportedOn:
+                            skipComponent = !supportedOn.pipelineTypes.Contains(currentPipelineType);
```

Comment:
So if you don't give an attribute, it will not be visible. That's probably a good choice, since it forces programmer to notice this attribute and use it, meaning we probably avoid more bugs where something only meant for SRP X accidentally ends up visible for SRP Y.